### ISSUE TITLE
Angular/cli 9.1 karma tests problem

### ISSUE THREAD

**Issue by KrakenTyio** [just-jeb/angular-builders#706]
## Describe the Bug
Hi Im upgrade to angular 9.1,
just now im using only karma tests with your custom-webpack + ts 3.7 and angular 9.0.
related to new angular release i have problem no matter if ts version 3.7 or 3.8.

Error what i have is in picture

![Screenshot from 2020-03-26 14-15-32](https://user-images.githubusercontent.com/5727376/77653403-e909ad80-6f6f-11ea-96f7-95ea285d5447.png)

nothing special in angular.json
only 
```
            "customWebpackConfig": {
              "path": "./builder/webpack.extras.ts"
            }
```

in extras is only `DefinePlugin` and added `externals`

**Comment by just-jeb** [comment]
If you switch the builder to a default one the tests pass?

**Comment by KrakenTyio** [comment]
if i switch back to default, test fail on not exist externals, but in deep look yes it will finish

On bottom of commnet is list of deps from package.json bc. i using many loaders etc. bc im not fully migrated all features to custom-webpack

weird situation is also when i remove `// @ts-ignore` from extras file which is config file for custom-webpack, test run down imediately bc. he doesn recognize `wco: Configuration, options: CustomWebpackBrowserSchema`

but this handle by ts-ignore

```ts
import { CustomWebpackBrowserSchema } from '@angular-builders/custom-webpack/dist';
import * as pkg from '../package.json';
import { Configuration, DefinePlugin } from 'webpack';

const buildUtil = require('./build-utils');

const config = require('../config/own.config').config.sandbox;

// @ts-ignore
export default (wco: Configuration, options: CustomWebpackBrowserSchema) => {
    wco = Object.assign(wco, {
        externals: buildUtil.externals
    } as Configuration);

    wco.plugins.push(
        new DefinePlugin({
            APP_VERSION: JSON.stringify(pkg.version),
            'process.env': {
                APP_CONFIG: JSON.stringify(config)
            }
        })
    );

    return wco;
};

```

![Screenshot from 2020-03-27 13-45-22](https://user-images.githubusercontent.com/5727376/77757286-4154b400-7031-11ea-91f8-5355b5f88711.png)


```json
"dependencies": {
    "@angular/animations": "~9.0.7",
    "@angular/common": "~9.0.7",
    "@angular/compiler": "~9.0.7",
    "@angular/core": "~9.0.7",
    "@angular/forms": "~9.0.7",
    "@angular/platform-browser": "~9.0.7",
    "@angular/platform-browser-dynamic": "~9.0.7",
    "@angular/router": "~9.0.7",
    "@angular/localize": "~9.0.7",
    "@agm/core": "~1.1.0",
    "@ckeditor/ckeditor5-build-classic": "~18.0.0",
    "@ngrx/effects": "~9.0.0",
    "@ngrx/store": "~9.0.0",
    "@ngx-translate/i18n-polyfill": "~1.0.0",
    "ckeditor": "~4.12.1",
    "core-js": "~3.6.4",
    "create-hmac": "~1.1.7",
    "croppie": "~2.6.4",
    "dwt": "~15.3.2",
    "moment": "~2.24.0",
    "moment-timezone": "~0.5.28",
    "ngx-clipboard": "~13.0.0",
    "pdfmake": "~0.1.65",
    "perfect-scrollbar": "~1.5.0",
    "resize-observer-polyfill": "~1.5.1",
    "rxjs": "~6.5.4",
    "sortablejs": "~1.10.2",
    "xlsx": "~0.15.6",
    "zone.js": "~0.10.3",
    "dom-autoscroller": "~2.3.4",
    "file-saver": "~2.0.2",
    "node-interval-tree": "~1.3.3",
    "tslib": "~1.11.1"
  },
  "devDependencies": {
    "@angular-builders/custom-webpack": "~9.0.0",
    "@angular-devkit/build-angular": "~0.900.7",
    "@angular-devkit/build-optimizer": "~0.900.7",
    "@angular/cli": "~9.0.7",
    "@angular/compiler-cli": "~9.0.7",
    "@angular/language-service": "~9.0.7",
    "@ngrx/store-devtools": "~9.0.0",
    "@ngtools/webpack": "~9.0.7",
    "@types/ckeditor": "~4.9.8",
    "@types/core-js": "~2.5.3",
    "@types/create-hmac": "~1.1.0",
    "@types/croppie": "~2.5.4",
    "@types/dwt": "~15.3.0",
    "@types/file-saver": "~2.0.1",
    "@types/googlemaps": "~3.39.3",
    "@types/hammerjs": "~2.0.36",
    "@types/jasmine": "~3.5.10",
    "@types/jasminewd2": "~2.0.8",
    "@types/karma": "~4.4.0",
    "@types/lodash": "~4.14.149",
    "@types/mocha": "~7.0.2",
    "@types/modernizr": "~3.5.3",
    "@types/new-relic-browser": "~0.1072.4",
    "@types/node": "~13.9.4",
    "@types/pdfmake": "~0.1.11",
    "@types/sortablejs": "~1.10.2",
    "@types/svg-sprite-loader": "~3.9.1",
    "@types/webpack-env": "~1.15.1",
    "@types/webpack-merge": "~4.1.5",
    "@types/youtube": "~0.0.38",
    "angular2-template-loader": "~0.6.2",
    "base64-js": "~1.3.1",
    "cache-loader": "~4.1.0",
    "codelyzer": "~5.2.2",
    "copy-webpack-plugin": "~5.1.1",
    "cross-env": "~7.0.2",
    "css-loader": "~3.4.2",
    "cssnano": "~4.1.10",
    "exports-loader": "~0.7.0",
    "file-loader": "~6.0.0",
    "fork-ts-checker-webpack-plugin": "~4.1.2",
    "hammerjs": "~2.0.8",
    "happypack": "~5.0.1",
    "html-loader": "~1.0.0",
    "html-webpack-plugin": "~4.0.2",
    "imports-loader": "~0.8.0",
    "istanbul-instrumenter-loader": "~3.0.1",
    "jasmine": "~3.5.0",
    "jasmine-core": "~3.5.0",
    "jasmine-spec-reporter": "~5.0.1",
    "junit": "~1.4.9",
    "karma": "~4.4.1",
    "karma-chrome-launcher": "~3.1.0",
    "karma-coverage-istanbul-reporter": "~2.1.1",
    "karma-jasmine": "~3.1.1",
    "karma-jasmine-html-reporter": "~1.5.3",
    "karma-junit-reporter": "~2.0.1",
    "karma-mocha-reporter": "~2.2.5",
    "karma-sourcemap-loader": "~0.3.7",
    "karma-webpack": "~4.0.2",
    "lodash": "~4.17.15",
    "mobile-detect": "~1.4.4",
    "moment-locales-webpack-plugin": "~1.2.0",
    "ng2-logger": "~3.0.11",
    "ngx-i18nsupport": "~0.17.1",
    "node-sass": "~4.13.1",
    "node-sass-tilde-importer": "~1.0.2",
    "normalize.css": "~8.0.1",
    "optimize-js-plugin": "~0.0.4",
    "parallel-webpack": "~3.0.0-alpha.2",
    "postcss": "~7.0.27",
    "postcss-loader": "~3.0.0",
    "postcss-preset-env": "~6.7.0",
    "postcss-url": "~8.0.0",
    "prettier": "~2.0.2",
    "protractor": "~5.4.3",
    "raw-loader": "~4.0.0",
    "rxjs-tslint-rules": "~4.29.2",
    "sass-loader": "~8.0.2",
    "script-ext-html-webpack-plugin": "~2.1.4",
    "source-map-loader": "~0.2.4",
    "stylelint": "~13.2.1",
    "stylelint-config-prettier": "~8.0.1",
    "stylelint-config-standard": "~20.0.0",
    "svg-sprite-loader": "~4.2.1",
    "svg-transform-loader": "~2.0.8",
    "svgo": "~1.3.2",
    "svgo-loader": "~2.2.1",
    "terser-webpack-plugin": "~2.3.5",
    "to-string-loader": "~1.1.6",
    "trust1connector": "~3.0.0",
    "ts-loader": "~6.2.2",
    "ts-node": "~8.8.1",
    "tslint": "~6.1.0",
    "tslint-config-prettier": "~1.18.0",
    "tslint-jasmine-rules": "~1.6.0",
    "typescript": "~3.7.5",
    "url-loader": "~4.0.0",
    "webpack": "~4.42.1",
    "webpack-bundle-analyzer": "~3.6.1",
    "webpack-cli": "~3.3.11",
    "webpack-dev-server": "~3.10.3",
    "webpack-merge": "~4.2.2"
  }
```

**Comment by KrakenTyio** [comment]
now im tested, and when i revert to `@angular-devkit/build-angular:karma` and remove connections to `externals` tests start and propagate to start testing, not checking code

i guess somewhere in process is ignored transpiler and tsc features introduced in 3.7

**Comment by KrakenTyio** [comment]
#708 its related, when i remove from your deps `@angular-devkit` and keep resolve by main host project its work well

**Comment by just-jeb** [comment]
Yep, indeed the same problem. All the builders [have](https://github.com/just-jeb/angular-builders/blob/master/packages/custom-webpack/package.json#L40) `^0.900.0` and `^9.0.0` dependencies in Angular DevKit and Angular Core respectively.  
The problem is that apparently `^0.900.0` doesn't match `0.901.0`. Therefore Angular Core is installed as expected (`^9.1.0` at the top level) but the devkit is not (two different versions - one is coming from Angular CLI 9.1 and the second from builder dependency).   

The easiest way to solve this is to release a new version of Angular Builders with minimal dependencies in `^0.901.0` and `^9.1.0`) but it will happen again in the next minor build so I have to check if there is another more robust solution.

**Comment by just-jeb** [comment]
Should be fixed in `v9.1.0-beta.1`. Please update me if helped.

**Comment by KrakenTyio** [comment]
yes its work well
